test/server/dataType.test.ts

```diff
@@ -0,0 +1,188 @@
+import { describe, expect, it } from "bun:test";
+import { toDataType, fromDataType } from "../../src/server/util";
+import type { TriggerType, JsonObject } from "../../src/types";
+
+describe("Data Type Conversion Functions", () => {
+  const trigger: TriggerType = "webhook";
+  
+  describe("toDataType", () => {
+    it("should handle null or undefined data", async () => {
+      const result = await toDataType(trigger, { data: null });
+      
+      expect(result.trigger).toBe(trigger);
+      expect(result.contentType).toBe("text/plain");
+      expect(result.payload).toBe("");
+    });
+    
+    it("should handle string data", async () => {
+      const data = "Hello, world!";
+      const result = await toDataType(trigger, { data });
+      
+      expect(result.trigger).toBe(trigger);
+      expect(result.contentType).toBe("text/plain");
+      expect(Buffer.from(result.payload, "base64").toString()).toBe(data);
+    });
+    
+    it("should handle object with contentType", async () => {
+      const data = {
+        contentType: "application/json",
+        base64: Buffer.from(JSON.stringify({ message: "Hello" })).toString("base64")
+      };
+      const result = await toDataType(trigger, { data });
+      
+      expect(result.trigger).toBe(trigger);
+      expect(result.contentType).toBe("application/json");
+      expect(result.payload).toBe(data.base64);
+    });
+    
+    it("should handle ArrayBuffer data", async () => {
+      const text = "Hello, world!";
+      const data = new TextEncoder().encode(text).buffer as ArrayBuffer;
+      const result = await toDataType(trigger, { data });
+      
+      expect(result.trigger).toBe(trigger);
+      expect(result.contentType).toBe("application/octet-stream");
+      expect(Buffer.from(result.payload, "base64").toString()).toBe(text);
+    });
+    
+    it("should handle Buffer data", async () => {
+      const text = "Hello, world!";
+      const data = Buffer.from(text);
+      const result = await toDataType(trigger, { data });
+      
+      expect(result.trigger).toBe(trigger);
+      expect(result.contentType).toBe("application/octet-stream");
```

Comment:
this could be a legit logic issue. 

can we check to see if its the default in the implementation?